test/src/tasks/publish-task.test.ts

```diff
@@ -91,9 +118,181 @@ describe(`Run publish task - ${testAppDir}`, function() {
+      const ganache = await startGanache(this.env)
+      closeGanache = ganache.close
+      await deployBases(this.env)
+    })
+
+    describe(`Repo does not exist, deploy new contract and repo for ${testAppDir}`, function() {
+      const bump: ReleaseType = 'major'
+
+      it('Run publish task', async function() {
+        const txData = await this.env.run(TASK_PUBLISH, {
+          bump,
+          // Manager must be the root account, to publish more test versions
+          managerAddress: rootAccount,
+          ipfsApiUrl
+        })
+
+        const {
+          to,
+          shortName,
+          managerAddress,
+          contractAddress,
+          version,
+          contentUri
+        } = parseNewRepoWithVersionTxData(txData)
+        // Compare with an object to see all of them at once
+        assert.deepEqual(
+          {
+            to,
+            shortName,
+            managerAddress,
+            version
+          },
+          {
+            to: defaultLocalAragonBases.apmAddress,
+            shortName: 'test',
+            managerAddress: rootAccount,
+            version: '1.0.0'
+          }
+        )
+
+        assert(
+          isAddress(contractAddress),
+          'Release contractAddress is not a valid address'
+        )
+
+        await assertContentUri(contentUri)
+
+        // Apply TX so it can be consumed by tests below
+        await broadcastTxData(txData, this.env)
+      })
+    })
+
+    describe('Repo exists, patch release with just content', function() {
+      const bump: ReleaseType = 'patch'
+      let repoState: RepoState
+
+      before('Fetch repo state', async function() {
+        repoState = await fetchRepoState(appName, this.env)
+      })
+
+      it('Run publish task', async function() {
+        const txData: PublishVersionTxData = await this.env.run(TASK_PUBLISH, {
+          bump,
+          ipfsApiUrl
+        })
+        await assertPublishTxData(repoState, bump, txData)
+
+        // Apply TX so it can be consumed by tests below
+        await broadcastTxData(txData, this.env)
+      })
+    })
+
+    describe('Repo exists, major release with contract and content', function() {
+      const bump: ReleaseType = 'major'
+      let repoState: RepoState
+
+      before('Fetch repo state', async function() {
+        repoState = await fetchRepoState(appName, this.env)
+      })
+
+      it('Run publish task', async function() {
+        const txData: PublishVersionTxData = await this.env.run(TASK_PUBLISH, {
+          bump,
+          ipfsApiUrl
+        })
+        await assertPublishTxData(repoState, bump, txData)
+
+        // Apply TX to check is valid
+        await broadcastTxData(txData, this.env)
+      })
+    })
+
+    after('Stop ganache instance', function() {
+      if (closeGanache) closeGanache()
+    })
   })
 
-  after('Stop ganache instance', function() {
-    if (closeGanache) closeGanache()
+  describe.skip('On the mainnet network', function() {
```

Comment:
Should we remove it or there is sections that makes sense to stay?

Reply:
Let's keep them and decide latter